### PR TITLE
Components :: create component FilterEditor

### PR DESCRIPTION
**Pull request by bloodstorms**
Create an intermediate component (`FilterEditor`) between `FilterStepForm` and `ConditionsEditor`.

`FilterEditor` contains conditionsEditor + filterSimpleCondition + all data transformation logic
to permit the export of the interface with all the its logical, but without filterStep env.

### CODE REVIEW

**Comment by bloodstorms** [comment]
[(8) ETQC à l'édition d'une permission, si j'ai choisi "can view specific rows", je peux filtrer la donnée avec le composant des conditions complexes](https://trello.com/c/MZ8tcuyo/107-8-etqc-%C3%A0-l%C3%A9dition-dune-permission-si-jai-choisi-can-view-specific-rows-je-peux-filtrer-la-donn%C3%A9e-avec-le-composant-des-condition)

File: src/components/stepforms/convert-filter-step-tree.ts [hunk 1]
```diff
@@ -142,29 +140,29 @@ export function buildFilterStepTree(conditionGroup: AbstractFilterTree, isRoot:
...
+  return filterStepGroup;
+
+  // if (isRoot) {
+  //   return {
+  //     name: 'filter',
+  //     condition: filterStepGroup,
+  //   };
+  // } else {
+  //   return filterStepGroup;
+  // }
```
Comment (davinov) [inline src/components/stepforms/convert-filter-step-tree.ts:152]: to remove?

File: src/components/stepforms/FilterEditor.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  filterTree!: FilterSimpleCondition | FilterComboAnd | FilterComboOr;
+
+  get conditionsTree() {
+    return buildConditionsEditorTree(this.filterTree);
+  }
+
+  updateFilterTree(newConditionsTree: AbstractFilterTree) {
+    const newFilterTree = buildFilterStepTree(newConditionsTree);
+    this.$emit('filterTreeUpdated', newFilterTree);
+  }
```
Comment (davinov) [inline src/components/stepforms/FilterEditor.vue:50]: Super klar :100:

File: src/components/stepforms/convert-filter-step-tree.ts (hunk 1 above)
Comment (bloodstorms) [reply src/components/stepforms/convert-filter-step-tree.ts:152]: woups, yes

**Pull request merged**